src/Data/Csv/Parser.hs

```diff
@@ -135,21 +155,30 @@ removeBlankLines = filter (not . blankLine)
 -- most likely want to use the 'endOfLine' parser in combination with
 -- this parser.
 record :: Word8  -- ^ Field delimiter
+       -> Bool   -- ^ Escape
        -> AL.Parser Record
-record !delim = V.fromList <$!> field delim `sepByDelim1'` delim
+record !delim !escape = V.fromList <$!> field delim escape `sepByDelim1'` delim
 {-# INLINE record #-}
 
--- | Parse a field. The field may be in either the escaped or
--- non-escaped format. The return value is unescaped.
-field :: Word8 -> AL.Parser Field
-field !delim = do
+-- | Parse a field. The escape option determines whether fields can be
+-- escaped with double quotes.
+field :: Word8  -- ^ Field delimiter
+      -> Bool   -- ^ Escape
```

Comment:
Maybe we can turn this Bool into an ADT, then one doesn't has to guess about the meaning of `True` and `False`.